### ISSUE TITLE
How to set aria-* attributes

### ISSUE THREAD

**Issue by ludmiloff** [yysun/apprun#21]
Hello @yysun, thank you for your effort on building this awesome framework.

The problem I faced is I'm not able to setup aria-* attributes, **aria-label** in particular with JSX/TSX syntax. What is strange, I'm able to successfully set **data-id** attribute which is quite similar. 
the code to reproduce:
`<form role="search"><input type="search" className="hsi" aria-label="Search"/></form>`
do not set aria-label attribute.

As I see, the jsx/react namespace is bound to apprun itself which correctly create element structure with all attributes, then somehow it is transformed into DOM element without aria-label;

Can you point me what is wrong.

UPDATE:
**vdom-my.ts
function updateProps(....)** 
just recognize some attributes like data-id, and tries to set others as attributes to DOM element, but desired behaviour should be to add the attribute to internal attributes map . I'm pretty sure I can prepare a pull request with desired code, at least with aria-* subset.

**Comment by yysun** [comment]
Thanks a lot!

**Comment by yysun** [comment]
I have added the unit tests. And released it to 1.13.1. I was going to
release a small fix in.d.ts file. Just bundled them.
Thank you again.

**Comment by yysun** [comment]
Checking name === 'role' in stead of startsWith is fine in this case, but the bigger problem is that other non-standard / custom attributes are still not support. I was thinking perhaps it should use _setAttribute_ in all cases instead of assigned the value to element property.
```javascript
for (let name in props) {
    const value = props[name];
    if (name === 'className') name = 'class';
    if (name === 'style') {
      if (element.style.cssText) element.style.cssText = '';
      for (let s in value) {
        if (element.style[s] !== value[s]) element.style[s] = value[s];
      }
    } else if (name.startsWith('data-')) {
      const dname = name.substring(5);
      if (element.dataset[dname] !== value) element.dataset[dname] = value;
    } else {
      // if (element[name] !== value) element[name] = value;
      if (element.getAttribute(name) !== value) element.setAttribute(name, value);
      if (name === 'key' && value) keyCache[value] = element;
    }
  }
```
However, I am not sure if there is any limitation and performance issue with _setAttribute_ , because the value parameter is alwsays string.

**Comment by ludmiloff** [comment]
Well, I did a preliminary test in chrome dev console (I'm not expert)
```
a = document.createElement("a")
<a>​</a>​
a.setAttribute("onlick", function(e) { console.log(e); })
undefined
a.getAttribute("onclick")
null
```
as it seems setAttribute will not work with clicks and such...

Currently I'm building a real world web app for my own with your wonderful framework and will report for any troubles and bugs I'm facing. 
What I mean is that we need more real world apps to collect enough use cases of HTML attributes.

Meanwhile, 
`if (element[name] !== value) element[name] = value;` is still necessary IMO

**Comment by kaveman-** [comment]
hello all.

checking out the framework and ran into the problem where custom attributes are ignored. 

I think setAttribute(...) is the correct way to set attributes.
@yysun if performance is a concern, the value can be checked for string types and then 'setAttribute'd.
@ludmiloff I tried your example above and it works for me. could it be the typo in attribute name that is failing you?

p.s: @yysun thanks for this neat little framework.

**Comment by kaveman-** [comment]
there seems to be a bug setting data-* attributes: only the 'data-' prefix is getting removed. IIUC, hyphenated-property-keys need to be camelCased as well. 

eg: 
`<div data-my-custom-prop='foo'>` fails to be set on the DOMStringMap raising an invalid property name exception.

**Comment by yysun** [comment]
That’s great! Thank you again. @kaveman- 
I will merge you fix.

**Comment by yysun** [comment]
The camel case for data- attributes fix is in v1.15.2
